File: Hangman/Python/Hangman.py
word = "banana"
table = []
index = 0
life = 5
game = True
for letter in word:
    table.append('_')

print(table)

while game:
    guess = input("A letter: ")

    if guess in word:
        for letter in word:
            if guess == letter:
                table[index] = guess

            index += 1
    else:
        life -= 1

    index = 0

    print(table)
    print(f"You have {life} chances remaining!")
    if (life == 0):
        game = False
        print(f"You lost! the word was {word}!")
        break
    if ("_" not in table):
        game = False
        print(f"You won! the word was {word}!")

File: Hangman/Ruby/hangman.rb
class Game
    def initialize(word = "banana")
        @word = word
        @len = @word.length
        @life = 5
        @game = true
        @guess = nil
        @table = Array.new(@len, "_")
    end

    def start_game
        puts "The secret word has #{@len} letters! You only have #{@life} lives!"
        while @game
            print @table.join(" ")
            print " Your Guess: "
            check_guess
            check_result
        end
    end

    def check_guess
        @guess = gets.chomp
        found = false

        for i in 0...@len
            if @guess == @word[i]
                @table[i] = @guess
                found = true
            end
        end

        unless found
            @life -= 1
            puts "Wrong!"
            puts "You have #{@life} guesses remaining"
            if @life == 0
                @game = false
                puts "You lost! The secret word is #{@word}"
            end
        end
    end

    def check_result
        unless @table.include?("_")
            @game = false
            puts "You won with #{@life} lives remaining!"
            puts "The word was #{@word}"
        end
    end
end

# ruby is so fun

game = Game.new
game.start_game
